libs/community/langchain_community/tools/todoist/tool.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+
+from langchain_core.tools import BaseTool
+from langchain_core.pydantic_v1 import Field
+
+from langchain_community.utilities.todoist import TodoistAPIWrapper
+
+
+class TodoistAction(BaseTool):
+    """A tool for interacting with Todoist."""
+
+    mode: str
+    name: str = "todoist"
+    description: str = "A wrapper around Todoist API."
+    api_wrapper: TodoistAPIWrapper = Field(default_factory=TodoistAPIWrapper)
+
+    def _run(self, *args, **kwargs) -> Any:
```

Comment:
i think we probably want to have separate Tool classes for each mode and explicitly enumerate the arguments to that API endpoint. this way those args are passed as part of the tool schema when using tool-calling apis

Reply:
Do you have an example that i can use? There doesnt seem to be a standard on how to define these tools